src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -55,16 +80,71 @@ Task IOpenInIDERequestHandler.ShowHotspotAsync(IShowHotspotRequest request)
                 throw new ArgumentNullException(nameof(request));
             }
 
+            return ShowHotspotAsync(request);
+        }
+
+        private async Task ShowHotspotAsync(IShowHotspotRequest request)
+        {
+            logger.WriteLine(OpenInIDEResources.OpenHotspot_ProcessingRequest, request.ServerUrl, request.ProjectKey,
+                request.OrganizationKey ?? OpenInIDEResources.OpenHotspot_NullOrganization, request.HotspotKey);
+
+            // Always show the Hotspots tool window. If we can't successfully process the
+            // request we'll show a gold bar in the window
+            toolWindowService.Show(HotspotsToolWindow.ToolWindowId);
+            failureInfoBar.Clear();
+
             if (!ideStateValidator.CanHandleOpenInIDERequest(request.ServerUrl, request.ProjectKey, request.OrganizationKey))
             {
-                return Task.CompletedTask;
+                // We're assuming the validator will have output an explanantion of why the IDE
+                // isn't in the correct state
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspot = await TryGetHotspotData(request.HotspotKey);
+            if (hotspot == null)
+            {
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspotViz = converter.Convert(hotspot);
```

Comment:
Now that I'm done with the converter I can say that it won't return a null, but it can throw an exception (e.g. if the SQ data is wrong, such as unsupported priority). I think it's best that the converter catches it and handles it. This way we're following the pattern that the callees are responsible for handling exceptions and the orchestrator only displays an infobar.
WDYT?
If you agree, I'll change the converter, otherwise we'll need to handle exceptions here.

Reply:
I don't have a strong preference, and the orchestrator is already handling exceptions from the Sonar web client.

Reply:
The converter doesn't currently log anything, and getting bad data from the server does seem like an exceptional condition. Maybe it's better if we keep the converter simple and let it throw.
I'll add exception handling to the orchestrator. We can always change it later if necessary.